powerauth-restful-security-spring-annotation/src/main/java/io/getlime/security/powerauth/rest/api/spring/model/ActivationContext.java

```diff
@@ -0,0 +1,254 @@
+package io.getlime.security.powerauth.rest.api.spring.model;
+
+import com.wultra.security.powerauth.client.v3.ActivationStatus;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+import java.util.List;
+
+/**
+ * Class representing the activation context data. It maps detailed activation attributes
+ * to a class that is supposed to be used by the developers in various scenarios.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+public class ActivationContext {
+    
+    private String activationId;
+    private String activationName;
+    private List<String> activationFlags;
+    private ActivationStatus activationStatus;
+    private String blockedReason;
+    private long applicationId;
+    private String userId;
+    private long version;
+    private XMLGregorianCalendar timestampCreated;
+    private XMLGregorianCalendar timestampLastUsed;
+    private XMLGregorianCalendar timestampLastChange;
+    private String platform;
+    private String deviceInfo;
+    private String extras;
+
+    /**
+     * Set activation ID.
+     * @param activationId Activation ID.
+     */
+    public void setActivationId(String activationId) {
+        this.activationId = activationId;
+    }
+
+    /**
+     * Get activation ID.
+     * @return Activation ID.
+     */
+    public String getActivationId() {
+        return activationId;
+    }
+
+    /**
+     * Set activation name.
+     * @param activationName Activation name.
+     */
+    public void setActivationName(String activationName) {
+        this.activationName = activationName;
+    }
+
+    /**
+     * Get activation name.
+     * @return Activation name.
+     */
+    public String getActivationName() {
+        return activationName;
+    }
+
+    /**
+     * Set activation flags.
+     * @param activationFlags Activation flags.
+     */
+    public void setActivationFlags(List<String> activationFlags) {
```

Comment:
I suggest to use `getActivationFlags().addAll()` or make a copy of `activationFlags` from request.